Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -440,8 +440,21 @@ AccessibilityRole AccessibilityNodeObject::determineAccessibilityRoleFromNode(Tr
         return AccessibilityRole::LandmarkMain;
     if (node()->hasTagName(navTag))
         return AccessibilityRole::LandmarkNavigation;
-    if (node()->hasTagName(asideTag))
+    // The aside element should not assume the complementary role when nested
+    // within the sectioning content elements.
+    // https://w3c.github.io/html-aam/#el-aside
+    if (node()->hasTagName(asideTag)) {
+        // Return LandmarkComplementary if the aside element explicitly has a role="complementary".
+        if (equalLettersIgnoringASCIICase(getAttribute(roleAttr), "complementary"_s))
```

Comment:
Using `ariaRoleAttribute() == AccessibilityRole::LandmarkComplementary` will likely be more robust here, since I believe that something like "   complementary" (note the whitespace) is still valid.

Reply:
Done.